.github/workflows/ci.yml

```diff
@@ -95,30 +185,142 @@ jobs:
       - name: Install Pulumi CLI
         uses: pulumi/action-install-pulumi-cli@releases/v1
       - uses: actions/checkout@v2
-      - name: Install Go dependencies
-        run: make ensure
-      - name: Run ${{ matrix.tests-set }} Tests
-        run: make specific_test_set TestSet=${{ matrix.tests-set }}
-  lint:
+      - name: Create test infra
+        run: |
+          export STACK=${{ secrets.PULUMI_TEST_OWNER }}/ci-cluster-$(head /dev/urandom | LC_CTYPE=C tr -dc '[:lower:]' | head -c5)
+          echo "::set-env name=STACK::$STACK"
+          make setup_test_infra
+
+  test-infra-destroy:
```

Comment:
I'm a little confused overall in what's going on in this file now. Is this step really about destroying?  Why do we have all the steps copied twice?  Why does `kubernetes` need a separate block?

Can we keep all this simpler?

Reply:
This *might* be able to get simpler - I might even break out Kubernetes to a different workflow file. The reason it's a separate block is that we want to be able to run AWS, GCP, Azure etc and at the same time, spin up our test infra for the Kubernetes tests

We can definitely do something to make this more obvious of course